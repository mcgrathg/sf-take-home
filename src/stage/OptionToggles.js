import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import OptionToggle from './OptionToggle';

const StyledOptions = styled.div`
  margin: 3rem 0;
`;

const OptionToggles = ({ options, selected, onOptionToggled }) => (
  <StyledOptions>
    {options.map(option => (
      <OptionToggle
        key={option}
        option={option}
        isSelected={Boolean(selected[option])}
        onToggled={onOptionToggled}
      />
    ))}
  </StyledOptions>
);

OptionToggles.propTypes = {
  onOptionToggled: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  selected: PropTypes.objectOf(PropTypes.bool).isRequired,
};

export default OptionToggles;
