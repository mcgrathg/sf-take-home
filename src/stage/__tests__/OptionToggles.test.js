import React from 'react';
import { render } from 'react-testing-library';
import { noOp } from 'test-utils';
import OptionToggles from '../OptionToggles';

test('renders', () => {
  const { container } = render(
    <OptionToggles options={[]} selected={{}} onOptionToggled={noOp} />,
  );

  expect(container.firstChild).toMatchSnapshot();
});
