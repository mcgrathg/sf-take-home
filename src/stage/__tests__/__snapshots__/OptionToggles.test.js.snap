// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.c0 {
  margin: 3rem 0;
}

<div
  class="c0"
/>
`;
